apm-sniffer/apm-sdk-plugin/mongodb-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mongodb/v3/define/v36/MongoDBInstrumentation.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.mongodb.v3.define.v36;
+
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import net.bytebuddy.matcher.ElementMatchers;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.ConstructorInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.InstanceMethodsInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.ClassInstanceMethodsEnhancePluginDefine;
+import org.apache.skywalking.apm.agent.core.plugin.match.ClassMatch;
+import org.apache.skywalking.apm.agent.core.plugin.match.NameMatch;
+
+import static org.apache.skywalking.apm.agent.core.plugin.bytebuddy.ArgumentTypeNameMatch.takesArgumentWithType;
+
+/**
+ * Enhance {@code com.mongodb.Mongo} instance, and intercept {@code com.mongodb.Mongo#createOperationExecutor(...)} method
+ * <p>
+ * support: 3.6.x
+ *
+ * @author dengliming
+ */
+public class MongoDBInstrumentation extends ClassInstanceMethodsEnhancePluginDefine {
+
+    private static final String ENHANCE_CLASS = "com.mongodb.Mongo";
+    private static final String METHOD_NAME = "createOperationExecutor";
+    private static final String INTERCEPTOR_CLASS = "org.apache.skywalking.apm.plugin.mongodb.v3.interceptor.v37.MongoDBClientDelegateInterceptor";
```

Comment:
A v37 interceptor for v36?

Reply:
I don't look carefully yet, I think there is missing `WITNESS_CLASS`?

Reply:
Yes, I think it is same treatment code, If special treatment is needed, I will move it to new package.

Reply:
> I don't look carefully yet, I think there is missing `WITNESS_CLASS`?
I can't find a class that can be witnessed for now,But "createOperationExecutor" this method it is called in 3.6

Reply:
If that method is not existing in other versions, it is safe. If they have different behaviors in different versions, then you need witness class.

Reply:
I conflict with `V3.0.x-3.5.x`, so we need to witness class.

Reply:
> I conflict with `V3.0.x-3.5.x`, so we need to witness class.

My understanding is that you suggest adding witness class to resolve the conflict between V3.0.x-3.5.x and V3.6?

Reply:
Conflict should be reviewed from codes level, rather than guess or through tests. The tests wouldn't cover all cases. If you found out instrumentation definition will trigger conflict, then witness class is required.

Reply:
@dmsolr Which version you tried will conflict,I just tried 3.4.0 works.

Reply:
@dmsolr Please confirm the confliction. It doesn't show up in the plugin CI tests.

Reply:
Sorry, I am late.
I notice that `Mongo` has been enhanced by `v30/MongoDBInstrumentation.java`. But they intercepted different methods. PLS, recheck it, maybe we can combine those Interceptor. (I will follow this too)

https://github.com/apache/skywalking/blob/f181f7b4830b05c6812878ccd4b23633163ea39a/apm-sniffer/apm-sdk-plugin/mongodb-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mongodb/v3/define/v30/MongoDBInstrumentation.java#L44-L50

Reply:
But v30/MongoDBInstrumentation.java provided an witness class，It only exist in version: 3.0.x~3.5.x.If we ombine those Interceptor,we need to remove the  witness class?

Reply:
But the witness class exists in v3.6.x too, right?

Reply:
Removing witness class is very dangerous, needs a little more deep dive. Because the old versions require that, then, it should have reasons.

Reply:
I think try to provide an witness class in v3.6 maybe a good solution？

Reply:
Yes, that is.

Reply:
I think so! I have a glance. it is compatible with v3.0-v3.7. (I am not very sure).

Reply:
I added witness class “com.mongodb.client.model.changestream.ChangeStreamDocument” for v3.6, This class only exist in version: 3.6.x or higher, This can solve the possible conflicts between V3.0.x-3.5.x and v3.6.x .(I think). See. https://mongodb.github.io/mongo-java-driver/3.6/whats-new/ @wu-sheng @dmsolr

Reply:
Got it.